syscall.go

```diff
@@ -115,10 +117,19 @@ func unmountShareDir() error {
 	return os.RemoveAll(containerMountDest)
 }
 
-func waitForBlockDevice(deviceName string) error {
-	devicePath := filepath.Join(devPath, deviceName)
+// Here in "0:0", the first number is the SCSI host number, while the second number is always 0.
+var scsiDiskPrefix = "/sys/class/scsi_disk/0:0:"
 
-	if _, err := os.Stat(devicePath); err == nil {
+func waitForBlockDevice(deviceName string, isSCSIAddr bool) error {
+	var path string
+
+	if isSCSIAddr {
+		path = scsiDiskPrefix + deviceName + "/device/block"
```

Comment:
This would be more consistent (and maybe safer) as:

```
path = filepath.Join(scsiDiskPrefix + deviceName, "/device/block")
```

You could also create a variable for the `/device/block` which is used multiple times.